pages/counter/[metricId].js

```diff
@@ -1,84 +1,67 @@
-import React from 'react'
-import { useRouter } from 'next/router'
-import Link from 'next/link'
-import { GraphQLClient } from 'graphql-request'
 import { Counter } from '@propeldata/react-counter'
-
-import { Layout } from '../../components'
+import { GraphQLClient } from 'graphql-request'
+import { useRouter } from 'next/router'
+import React from 'react'
+import { Layout, Loader } from '../../components'
 import { MetricQuery } from '../../graphql'
+import { useDataFetching } from '../../utils'
 
 const client = new GraphQLClient(
   process.env.NEXT_PUBLIC_GRAPHQL_ENDPOINT_US_EAST_2
 )
 
 export default function CounterPage() {
-  const [isLoading, setIsLoading] = React.useState(true)
   const [uniqueName, setUniqueName] = React.useState()
   const [description, setDescription] = React.useState()
-  const [accessToken, setAccessToken] = React.useState()
 
   const router = useRouter()
   const { metricId } = router.query
 
-  React.useEffect(() => {
-    async function fetchData() {
-      try {
-        setIsLoading(true)
-        const accessToken = window.localStorage.getItem('accessToken')
-        setAccessToken(accessToken)
-        client.setHeader('Authorization', `Bearer ${accessToken}`)
-        const response = await client.request(MetricQuery, { metricId })
-        const metric = response?.metric
-
-        setUniqueName(metric?.uniqueName)
-        setDescription(metric?.description)
-      } catch (error) {
-        console.log(error)
-      } finally {
-        setIsLoading(false)
-      }
-    }
+  const { isLoading, accessToken, metric } = useDataFetching(
+    client,
+    metricId,
+    MetricQuery
+  )
```

Comment:
This abstraction is great, I think we can improve it by passing only the `metricId` and let the hook create the client instance and import the MetricQuery directly.

```suggestion
  const { isLoading, accessToken, metric } = useDataFetching(metricId)
```

Reply:
Done!